website/docs/tutorials/Guides/testing.mdx

```diff
@@ -1,7 +1,217 @@
+```
+# src/storage_contract.cairo
+
+%lang starknet
+%builtins pedersen range_check
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+# Define a storage variable.
+@storage_var
+func balance() -> (res : felt):
+end
+
+# Increases the balance by the given amount.
+@external
+func increase_balance{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(amount: felt):
+    let (res) = balance.read()
+    balance.write(res + amount)
+    return ()
+end
+
+# Returns the current balance.
+@view
+func get_balance{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+        res : felt):
+    let (res) = balance.read()
+    return (res)
+end
+
+@constructor
+func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    balance.write(0)
+    return ()
+end
+```
+Then define a proxy contract, your test will call storage contract using proxy contract
+```
+# src/proxy_contract.cairo
+
+%lang starknet
+%builtins pedersen range_check
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+
+@contract_interface
+namespace StorageContract:
+    func increase_balance(amount : felt):
+    end
 
-T.B.D.
+    func get_balance() -> (res : felt):
+    end
+end
+
+@storage_var
+func target() -> (res : felt):
+end
+
+# Sets a basic contract address which proxy calls
+@external
+func set_target{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(new_target: felt):
+    target.write(new_target)
+    return ()
+end
+
+# Invokes increase_balance method
+@external
+func increase_balance{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(amount: felt):
+    let (target_contract) = target.read()
+    StorageContract.increase_balance(contract_address=target_contract, amount=amount)
+    return ()
+end
+
+
+# Calls get_balance method and returns the result
+@view
+func get_balance{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+        res : felt):
+    let (target_contract) = target.read()
+    let (res) = StorageContract.get_balance(contract_address=target_contract)
+    return (res)
+end
+```
+At the end define testing contract with a single test case
+```
+# tests/test_proxy.cairo
+%lang starknet
+
+@contract_interface
+namespace ProxyContract:
+    func set_target(new_target : felt):
+    end
+
+    func increase_balance(amount : felt):
+    end
+
+    func get_balance() -> (res : felt):
+    end
+end
+
+@external
+func test_proxy_contract{syscall_ptr : felt*, range_check_ptr}():
+    alloc_locals
+
+    local contract_logic_address : felt
+    local contract_proxy_address : felt
+    %{
+        # We deploy two contracts and put their addresses into local variables
+        ids.contract_proxy_address = deploy_contract("./src/proxy_contract.cairo").contract_address 
+        ids.contract_logic_address = deploy_contract("./src/storage_contract.cairo").contract_address 
+    %}
```

Comment:
This is the key takeaway from this section. I'd hide the remaining code in the one of two ways:
- approach used by docusaurus to "Github action files"   https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions
- create examples directory in root repo and move it there